packages/react-async/src/useAsync.tsx

```diff
@@ -282,10 +273,10 @@ function isEvent(e: FetchRunArgs[0]): e is Event | React.SyntheticEvent {
  * @param {FetchOptions} options
  * @returns {AsyncState<T, FetchRun<T>>}
  */
-function useAsyncFetch<T>(
+function useAsyncFetch<T, C>(
```

Comment:
The typing for this function is kinda wack.

First of all, the `init` parameter is using a `RequestInit` type that doesn't seem to actually exist. I think the type parameter `C` here should replace every instance of `RequestInit`... or rather, perhaps `C` should just be renamed to `RequestInit` here?

Second, the 3rd argument's type probably shouldn't be `FetchOptions`, but rather a subset of `FetchOptions` (that may not need the 2nd type parameter `C` at all), because passing in an object containing `promiseFn` or `deferFn` props doesn't make any sense in this context, yet it's technically allowed by the types here.

Reply:
I found `RequestInit` defined in `node_modules/@types/react-native/globals.d.ts`. Appears to be a global type which should explain why it isn't declared in this file.
And I think C works just fine here as it's the type for the **C**ontext rather than the inits. So, it probably shouldn't replace any instance of `RequestInit`.